### PR TITLE
AzureMonitor -  E2E tests drone update

### PR DESCRIPTION
**Pull request by aangelisc**
This PR adds Azure Monitor E2E tests to drone. These tests should only run on PR's from the Grafana repo (not forks).

This update is dependent on adding variables to the drone configuration for Grafana.

It may be prudent to add the `failure: ignore` flag or to limit these tests to only run on PR's related to the Azure Monitor data-source for now to determine if they're flaky or not due to the dependency on resource deployment and the interaction with the Azure APIs.

grafana/grafana#54619 needs to be merged before this PR.

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -492,6 +492,39 @@ steps:
     HOST: grafana-server
   image: cypress/included:9.5.1-node16.14.0-slim-chrome99-ff97
   name: end-to-end-tests-various-suite
+- commands:
+  - apt-get update && apt-get install git curl jq --yes
```
Comment (andresmgot) [inline .drone.yml:496]: maybe you can skip the `update`?

File: .drone.yml [hunk 2]
```diff
@@ -492,6 +492,39 @@ steps:
...
+  - cd ..
+  - git clone "https://$${GITHUB_TOKEN}@github.com/grafana/cloud-data-sources" --depth=1
+  - cd cloud-data-sources/e2e-scripts
+  - ./scripts/install-dependencies-ci.sh azure
```
Comment (andresmgot) [inline .drone.yml:500]: maybe we can skip the `cd`, I don't think there is a problem to clone the repo in the current working directory (and call the scripts as `./cloud-data-sources/e2e-scripts/scripts/install-dependencies-ci.sh`)

File: .drone.yml [hunk 3]
```diff
@@ -492,6 +492,39 @@ steps:
...
+  - export PATH=$PATH:/root/.pulumi/bin
+  - ./scripts/azure-login.sh
+  - ./scripts/install.sh azure
+  - ./scripts/azure-pulumi-login.sh azure deploy ${DRONE_SOURCE_BRANCH}
+  - cd ../../grafana
+  - ./bin/build e2e-tests --port 3001 --suite cloud-plugins-suite --video false
+  - cd ../cloud-data-sources/e2e-scripts
+  - ./scripts/destroy.sh azure destroy ${DRONE_SOURCE_BRANCH}
```
Comment (andresmgot) [inline .drone.yml:508]: we need to make sure that this command runs even if the previous (`./bin/build e2e-tests`) fails, is there any safeguard for that?

File: .drone.yml [hunk 4]
```diff
@@ -1284,6 +1317,39 @@ steps:
     HOST: grafana-server
   image: cypress/included:9.5.1-node16.14.0-slim-chrome99-ff97
   name: end-to-end-tests-various-suite
+- commands:
```
Comment (andresmgot) [inline .drone.yml:1320]: curious, why is this defined twice?

**Review by andresmgot — COMMENTED** [review]
Awesome!

> It may be prudent to add the failure: ignore flag or to limit these tests to only run on PR's related to the Azure Monitor data-source for now to determine if they're flaky or not due to the dependency on resource deployment and the interaction with the Azure APIs.

I agree. I would either make a explicit check (e.g. only run if the label  `datasource/Azure` is present) or automatic (if  `pkg/tsdb/azuremonitor` or `public/app/plugins/datasource/grafana-azure-monitor-datasource` has been modified). Probably start with the first option while we make it stable.

File: .drone.yml (hunk 1 above)
Comment (aangelisc) [reply .drone.yml:496]: Most likely yep, it's a force of habit to run update before install. If this proves to be stable then my plan for the future would be to bundle the image ourselves to make the E2E tests more efficient.

File: .drone.yml (hunk 2 above)
Comment (aangelisc) [reply .drone.yml:500]: I was doing this as `yarn` was having issues if the repo was cloned into the Grafana directory, but I can make this work 😊

File: .drone.yml (hunk 3 above)
Comment (aangelisc) [reply .drone.yml:508]: I will have a think on how to implement this, good catch!

File: .drone.yml (hunk 4 above)
Comment (aangelisc) [reply .drone.yml:1320]: Once for PRs and once for main.

File: .drone.yml [hunk 5]
```diff
@@ -492,6 +492,39 @@ steps:
...
```
Comment (asimpson) [inline .drone.yml:495]: Would these steps be easier to manage in an [image](https://docs.drone.io/pipeline/docker/syntax/images/) we build for this purpose? Besides controlling updates better we could also cut down the boilerplate here. Heck, we could use `cloud-data-sources` as the package repo for such an image.

**Review by asimpson — COMMENTED** [review]
This looks really good, nice work! I only had the one question and I agree with @andresmgot's call out re: tearing stuff down in case of errors.

File: .drone.yml (hunk 5 above)
Comment (aangelisc) [reply .drone.yml:495]: That's a great shout, will implement this as it'll simplify things massively!

**Review by dsotirakis — APPROVED** [review]
Awesome! 🎉

**Review by andresmgot — APPROVED** [review]
:rocket:

**Pull request merged**